daemons/controld/controld_execd.c

```diff
@@ -1917,59 +1917,75 @@ construct_op(lrm_state_t * lrm_state, xmlNode * rsc_op, const char *rsc_id, cons
         op->interval_ms = 0;
     }
 
-#if ENABLE_VERSIONED_ATTRS
-    // Resolve any versioned parameters
-    if (lrm_state && pcmk__str_none_of(op->op_type, RSC_METADATA, CRMD_ACTION_DELETE, NULL)
-        && !is_remote_lrmd_ra(NULL, NULL, rsc_id)) {
-
-        // Resource info *should* already be cached, so we don't get executor call
+    if (lrm_state) {
         lrmd_rsc_info_t *rsc = lrm_state_get_rsc_info(lrm_state, rsc_id, 0);
-        struct ra_metadata_s *metadata;
-
-        metadata = metadata_cache_get(lrm_state->metadata_cache, rsc);
```

Comment:
Let's split this into two commits, the first should break apart the if and functionize the versioned attributes stuff so it doesn't clutter this function, then the actual changes will be more obvious in the second commit

Reply:
Ack, will do.